src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -120,7 +109,11 @@ protected override bool ReleaseHandle()
             // to retry, as the descriptor could actually have been closed, been subsequently reassigned, and
             // be in use elsewhere in the process.  Instead, we simply check whether the call was successful.
             int result = Interop.Sys.Close(handle);
-            Debug.Assert(result == 0, $"Close failed with result {result} and error {Interop.Sys.GetLastErrorInfo()}");
+            if (result != 0)
+            {
+                t_lastCloseErrorInfo = Interop.Sys.GetLastErrorInfo();
+                Debug.Fail($"Close failed with result {result} and error {t_lastCloseErrorInfo}");
```

Comment:
I wonder whther a test is feasible (that of course would require this assert go away)

Reply:
I'm trying to think of a way to reach this code in a unit test, that does not involve having a remote drive with little space left.

Reply:
I was thinking of a ramdisk:
```sh
$ mkdir /tmp/ram
$ sudo mount -t tmpfs -o size=1k ram /tmp/ram
$ cp runtime/CODE-OF-CONDUCT.md /tmp/ram/worksok
$ cp runtime/THIRD-PARTY-NOTICES.TXT /tmp/ram/fails
cp: error writing '/tmp/ram/fails': No space left on device
```
but, (1) that needs sudo (2) it's probably a lot of syscalls. 

@tmds, can you think of an easy way for a test to simulate a very small disk, or some other way to fail one write?

Reply:
There is [/dev/full](https://en.wikipedia.org/wiki//dev/full). It appears to be available on linux:

```
$ echo asdf > /dev/full
-bash: echo: write error: No space left on device
```

Reply:
Thanks, that works unless he needs the first write to succeed.

Reply:
I wrote a little test program and `/del/full` does not return an error for `close`:

```c
#include <unistd.h>
#include <stdio.h>
#include <fcntl.h>

int
main(int argc, char** argv)
{
        int fd = open("/dev/full", O_WRONLY | O_TRUNC);
        if (fd == -1)
        {
                perror("open");
                return 1;
        }
        if (close(fd))
        {
                perror("close");
                return 2;
        }
        puts("no errors");
        return 0;
}
```

Prints "no errors". So `/dev/full` might not be as useful as I thought.

Reply:
Thanks @AustinWise for the suggestion, but unfortunately, as Dan mentioned, the key to get this to repro is to have at least one successful write. If I'm not mistaken, `/dev/full` would always fail.

Reply:
If an automated test is impractical, that's shouldn't stop this change 😀. Maybe a manual test [like we have in Console](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Console/tests/ManualTests/ManualTests.cs) is enough.

Reply:
Interesting, thanks for the suggestion! I didn't know we had official manual tests. I will definitely add one.
I'm also exploring if using `/dev/shm` could help. I can create a temporary file system in RAM, just like you did with `/tmp/ram`, but this option does not seem to require super user privileges.

Reply:
Ok I added a manual test. @jkotas can you please take a look at this PR?